share/wake/html/main.js

```diff
@@ -197,11 +197,36 @@ function workspace(node) {
   return res;
 }
 
+function createThemeToggle() {
+  const prefersDark = matchMedia("prefers-color-scheme: dark");
```

Comment:
Can you add support for localStorage that way the site remembers when someone edits the toggle? It should only be a few lines of code. https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage#examples

My personal preference is dark theme but (I think) I haven't bothered to setup my OS preference so it would keep defaulting to light theme

Reply:
Yeah, I was debating whether or not I should use local storage for persisting the choice, so I can go add that now.

One thing to note is that if you're opening the file locally, with a `file://` protocol, then there will probably be issues with using local storage for security reasons. local storage should work fine for anything that's actually hosted off an actual HTTP server, though. I can test out a few scenarios locally to see what happens and report back on the behavior.

Reply:
Interestingly, it looks like different browsers do different things. In both Firefox and Safari, for `file://` protocols, each individual file appears to have its own, private localStorage. Chrome, however, appears to share the same localStorage across multiple files.

For anything served off HTTP(S), all pages under the same domain and port number share the same localStorage.

Anyway, I pushed up a change that now saves the user's preference to localStorage: https://github.com/sifive/wake/pull/1180/commits/2a68be780d0012c4b46d76375eb8660d5161d31a